engine/src/render/shadow_atlas_rasterizer/plugin.cpp

```diff
@@ -208,5 +236,105 @@ void cubos::engine::shadowAtlasRasterizerPlugin(Cubos& cubos)
+                              static_cast<int>(slot->offset.y * float(atlas.getPointAtlasSize().y)),
+                              static_cast<int>(slot->size.x * float(atlas.getPointAtlasSize().x)),
+                              static_cast<int>(slot->size.y * float(atlas.getPointAtlasSize().y)));
+
+                // Send the PerScene data to the GPU.
+                PerSceneCube perScene;
+                auto proj = glm::perspective(glm::radians(90.0F),
+                                             (float(atlas.getPointAtlasSize().x) * slot->size.x) /
+                                                 (float(atlas.getPointAtlasSize().y) * slot->size.y),
+                                             0.1F, light.range);
+
+                for (int i = 0; i < 6; i++)
+                {
+                    glm::mat4 view;
+                    switch (i)
+                    {
+                    case 0:
+                        view = glm::inverse(glm::scale(localToWorld.mat, glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    case 1:
+                        view = glm::inverse(
+                            glm::scale(glm::rotate(localToWorld.mat, glm::radians(180.0F), glm::vec3(0.0F, 1.0F, 0.0F)),
+                                       glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    case 2:
+                        view = glm::inverse(
+                            glm::scale(glm::rotate(localToWorld.mat, glm::radians(90.0F), glm::vec3(0.0F, 1.0F, 0.0F)),
+                                       glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    case 3:
+                        view = glm::inverse(
+                            glm::scale(glm::rotate(localToWorld.mat, glm::radians(-90.0F), glm::vec3(0.0F, 1.0F, 0.0F)),
+                                       glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    case 4:
+                        view = glm::inverse(
+                            glm::scale(glm::rotate(localToWorld.mat, glm::radians(90.0F), glm::vec3(1.0F, 0.0F, 0.0F)),
+                                       glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    case 5:
+                        view = glm::inverse(
+                            glm::scale(glm::rotate(localToWorld.mat, glm::radians(-90.0F), glm::vec3(1.0F, 0.0F, 0.0F)),
+                                       glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    default:
+                        view = glm::inverse(glm::scale(localToWorld.mat, glm::vec3(1.0F / localToWorld.worldScale())));
+                        break;
+                    }
+
+                    perScene.lightViewProj[i] = proj * view;
+                }
```

Comment:
This is the same exact math as before right? In that case, it would probably be a good idea to move it to a common place, so that you don't define it twice